### PR TITLE
Split controller config out from api server config

### PR DESCRIPTION
**Pull request by smarterclayton**
During startup of the controller, do not start RESTConfigs for master processes by splitting the config path when running in "controller only" mode. Avoids keeping a large chunk of things in memory that have no business being in memory (6 rest storage, which bring along caches)

@deads2k @liggitt [test]

### CODE REVIEW

File: pkg/cmd/server/kubernetes/master/master_config.go [hunk 1]
```diff
@@ -642,6 +642,42 @@ func BuildKubernetesMasterConfig(
...
+	if err != nil {
+		return nil, err
+	}
+
+	schedulerServer, err := buildSchedulerServer(masterConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kmaster := &MasterConfig{
```
Comment (deads2k) [inline pkg/cmd/server/kubernetes/master/master_config.go]: I'd rather see a new struct built that contains what we need and build that.  Slimming down is never done and side-effects get weird.

**Comment by smarterclayton** [comment]
I don't have time to do that and it would ripple too much.  That's
easily something that can be done next release.

File: pkg/cmd/server/origin/master.go [hunk 2]
```diff
@@ -250,23 +247,11 @@ func (c *MasterConfig) RunHealth() error {
...
-	}
-	sarClient := clientgoclientset.New(c.PrivilegedLoopbackKubernetesClientsetInternal.Authorization().RESTClient()).AuthorizationV1beta1().SubjectAccessReviews()
-	remoteAuthz, err := authzwebhook.NewFromInterface(sarClient, 5*time.Minute, 5*time.Minute)
-	if err != nil {
-		return err
-	}
-
 	// we use direct bypass to allow readiness and health to work regardless of the master health
-	authz := serverhandlers.NewBypassAuthorizer(remoteAuthz, "/healthz", "/healthz/ready")
+	authz := serverhandlers.NewBypassAuthorizer(c.Authorizer, "/healthz", "/healthz/ready")
```
Comment (deads2k) [inline pkg/cmd/server/origin/master.go]: I think this is only used by the controllers, but the construct built here where authorizer can mean different things is really undesireable.

**Comment by deads2k** [comment]
I am -1 on this change.  It makes it nearly impossible to figure out what fields like `.Authorizer` mean depending on context.  "What is *this* `MasterConfig` for?" isn't a question I should have to ask.  I would fear our bugs.

**Comment by smarterclayton** [comment]
The master config is for configuring a master that does three
overlapping Venn diagrams of things.

You need to suggest an alternative that is containable today that
doesn't increase risk if you want something different.  I agree
separation is good, but it's not worse than today, and current state
is unacceptable.

**Comment by deads2k** [comment]
> You need to suggest an alternative that is containable today that
> doesn't increase risk if you want something different.

Let's get @mfojtik's change in and then just plumb it properly  He'll have functions to build the init functions and we'll use one to lead to the other with distinct types.

**Comment by smarterclayton** [comment]
@mfojtik you are now on my critical path

**Pull request closed without merging**